service/service.go

```diff
@@ -56,10 +56,17 @@ var getObservationStore = func(ctx context.Context) (observationStore *graph.DB,
 	return observationStore, errorConsumer, nil
 }
 
+var pConfig *kafka.ProducerConfig
+
+var cfg *config.Config
+
 // getProducer returns a kafka producer
-var getProducer = func(ctx context.Context, kafkaBrokers []string, topic string, envMax int) (kafkaProducer kafka.IProducer, err error) {
+var getProducer = func(ctx context.Context, kafkaBrokers []string, topic string) (kafkaProducer kafka.IProducer, err error) {
```

Comment:
`getProducer` should accept a config, which is then used to create pConfig.
maybe the signature can be:
`func(ctx context.Context, cfg *config.Config, topic string)`
and then you can obtain the values for kafkaVersion (as you do now), as well as kafkaBrokers and maxBytes